core/emulate.c

```diff
@@ -403,6 +417,8 @@ static em_status_t operand_read(struct em_context_t *ctxt,
         if (op->flags & OP_READ_PENDING) {
             rc = ctxt->ops->read_memory_post(ctxt->vcpu, &op->value, op->size);
         } else {
+            if (ctxt->opcode.flags & INSN_BITOP)
```

Comment:
nit: This works, but kind of "hacks" into the usual `em_emulate_insn()` workflow: parsing of the destination operand ends up taking care of the next/source operand. I think it hurts readability.

Wouldn't it be better to place this logic somewhere between `decode_operands()` and the first `operand_read()` call?

Reply:
True, that's much better. I'll apply your suggestion when I'm back next weekend.